src/cascade/model/grids.py

```diff
@@ -0,0 +1,235 @@
+"""Support for dismod concepts that are expressed over grids: age-time, smoothings and value priors.
+"""
+from math import isclose, isinf
+
+import numpy as np
+
+GRID_SNAP_DISTANCE = 1 / 365
+
+
+def _nearest(values, query):
+    values = np.array(values)
+    return values[np.abs(values - query).argmin()]
+
+
+class AgeTimeGrid:
+    """Structure for storing age and times over which the model will estimate.
+    Also constructors for making standard age-time grids.
+    """
+
+    @classmethod
+    def uniform(cls, age_start, age_end, age_step, time_start, time_end, time_step):
+        """Construct an age-time grid with uniform spacing on each axis.
+        All values are in units of years.
+
+        Args:
+            age_start: the lowest age in the grid
+            age_end: the highest age in the grid
+            age_step: the spacing of age points
+            time_start: the earliest time in the grid
+            time_end: the latest time in the grid
+            time_step: the spacing of time points
+        """
+        ages = np.arange(age_start, age_end, age_step)
+        times = np.arange(time_start, time_end, time_step)
+
+        return cls(ages, times)
+
+    def __init__(self, ages, times, snap_distance=GRID_SNAP_DISTANCE):
+        self._ages = tuple(sorted(set(ages)))
```

Comment:
This set should take snap distance into account. That's a complicated maneuver. Maybe that's a later feature. For instance, if two values are within 1/365 of each other, and you have to set a value, which of the two do you choose? What if one value is near two others, but those two are further than 1/365 away from the first? Do we snap all values to the closest that's 1/365? Meaning multiply by 365, round them, convert to int, and then make the set?

Reply:
Good point. Maybe the right thing to do is reduce the snap distance down to floating point error and call it good? That doesn't actually resolve your question but it makes it unlikely to be an issue in practice.

Reply:
Yes, wait. Is it a problem if two ages are nearly the same but not quite? .... I don't think it is. Nice!

Reply:
I'm realizing that if you have grids where the points aren't at integer values or simple floats and the snap distance is small then this gets awkward to work with. Maybe that won't matter in practice.